File: chat-room-common/src/main/java/io/github/wj0410/chatroom/common/message/PromptMessage.java
package io.github.wj0410.chatroom.common.message;

import io.github.wj0410.chatroom.common.enums.old.PromptType;
import lombok.Data;

/**
 * @author wangjie
 * @date 2023/10/26
 */
@Data
public class PromptMessage {
    private PromptType promptType;
    private String clientId;
    private String msg;
    private long timestamp;


    public PromptMessage(){
        this.timestamp = System.currentTimeMillis();
    }

    @Override
    public String toString() {
        return "PromptMessage{" +
                "promptType=" + promptType +
                ", clientId='" + clientId + '\'' +
                ", msg='" + msg + '\'' +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: chat_vue/src/model/PromptMessage.ts
import {MessageType, PromptType} from "../constant/Enum";
import {Message} from "./Message";

export class PromptMessage {
    promptType: PromptType
    clientId: string
    msg: string
    timestamp: bigint

    constructor(data) {
        this.promptType = data.promptType;
        this.clientId = data.clientId;
        this.msg = data.msg;
        this.timestamp = data.timestamp;
    }

    buildMessage() {
        let obj = {
            type: MessageType.PROMPT,
            data: this
        }
        return new Message(obj);
    }
}
